pkg/logs/sender/sender.go

```diff
@@ -66,20 +69,30 @@ func (s *Sender) run() {
 
 // send sends a payload to multiple destinations,
 // it will forever retry for the main destination unless the error is not retryable
-// and only try once for additionnal destinations.
+// and only try once for additional destinations.
 func (s *Sender) send(payload []byte) error {
 	for {
 		err := s.destinations.Main.Send(payload)
 		if err != nil {
+			if s.lastError == nil {
+				s.hasError <- true
```

Comment:
I might be missing something obvious, but where is this channel drained when no backup endpoint is defined?

Reply:
Good catch - this was not accounted for. 
Ive made some changes to the sender to handle this better and added a UT to ensure both cases handle and recover from errors correctly.